From ac6010448ba29f8c5b979d11cabeb09a91cf260c Mon Sep 17 00:00:00 2001
From: Mikhail Novosyolov <m.novosyolov@rosalinux.ru>
Date: Mon, 31 Aug 2020 23:44:17 +0300
Subject: [PATCH] Recognize systemd.unit=anaconda.target in anaconda-generator

anaconda.target may be activated not only by symlinking /etc/systemd/systemd/default.target,
this symlink may even not exist, but by e.g. adding "systemd.unit=anaconda.target"
to kernel cmdline (see kernel-command-line(7) from systemd)

`systemctl is-active -q anaconda.target` could be used here, but systemctl is not available from systemd generators.

P.S. See https://www.redhat.com/archives/anaconda-devel-list/2012-August/msg00118.html for description why generator is needed.
---
 data/systemd/anaconda-generator | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/systemd/anaconda-generator b/data/systemd/anaconda-generator
index cda87480d..bafe74ea1 100755
--- a/data/systemd/anaconda-generator
+++ b/data/systemd/anaconda-generator
@@ -5,7 +5,7 @@
 ANACONDA_TARGET="/lib/systemd/system/anaconda.target"
 CURRENT_DEFAULT_TARGET=$(readlink /etc/systemd/system/default.target)
 
-if [ "$ANACONDA_TARGET" != "$CURRENT_DEFAULT_TARGET" ]; then
+if ! { [ "$ANACONDA_TARGET" = "$CURRENT_DEFAULT_TARGET" ] || grep -q 'systemd.unit=anaconda.target' /proc/cmdline ;} ; then
     exit 0
 fi
 
-- 
2.23.0

